imcsdk/imccore.py

```diff
@@ -143,7 +143,7 @@ def __init__(self, class_id, tag_name=None):
         self._tag_name = tag_name
         ImcBase.__init__(self, class_id)
 
-    def to_xml(self, xml_doc=None, option=None, elem_name=None):
+    def to_xml(self, xml_doc=None, option=None, elem_name=None, cookie=None):
```

Comment:
`cookie`  is present in the imcmethod class object. 
With reference to the xml element tree, the filter object will be a child object of the imcmethod object.
But, with reference to inheritance, the imcmethod and filter class are siblings, deriving from the ImcBase class.
Hence, the cookie from `self.cookie` will not get automatically inherited in the filter object.
The reason for making the cookie available in the filter class is if we want to pro-actively catch non-applicable properties for a particular platform even before doing a query_dn/query_classid on the server.